lib/supports/diffman.rb

```diff
@@ -0,0 +1,53 @@
+class Diffman
+  def initialize
+    @tempfiles = []
+  end
+
+  def diff(string1, string2)
+    files = [tempfile(string1), tempfile(string2)]
+    params = files.map(&:path)
+    Open3.popen3(diff_bin, *params) { |i, o, e, w| o.read }
+  ensure
+    remove_tempfiles!
+  end
+
+  def patch(string, patch, reverse: false)
+    files = [tempfile(string), tempfile(patch)]
+    params = files.map(&:path)
+    params << '-R' if reverse
+    return nil if Open3.popen3(patch_bin, *params) { |i, o, e, w | w.value } != 0
```

Comment:
Unused block argument - `i`. If it's necessary, use `_` or `_i` as an argument name to indicate that it won't be used.<br>Unused block argument - `o`. If it's necessary, use `_` or `_o` as an argument name to indicate that it won't be used.<br>Unused block argument - `e`. If it's necessary, use `_` or `_e` as an argument name to indicate that it won't be used.<br>Space after last block parameter detected.